src/ts/edit.ts

```diff
@@ -59,14 +59,23 @@ Dropzone.options.elabftwDropzone = {
     });
 
     // once it is done
-    this.on('complete', function(answer: any) {
+    this.on('complete', function(answer: any) { 
       // check the answer we get back from app/controllers/EntityController.php
       const json = JSON.parse(answer.xhr.responseText);
       notif(json);
       // reload the #filesdiv once the file is uploaded
       if (this.getUploadingFiles().length === 0 && this.getQueuedFiles().length === 0) {
         $('#filesdiv').load('?mode=edit&id=' + $('#info').data('id') + ' #filesdiv', function() {
           displayMolFiles();
+	var dropZone = Dropzone.forElement('#elabftw-dropzone');
+	
+	// Check to make sure the success function is set by tinymce and we are dealing with an image drop and not a regular upload
+	  if (typeof dropZone.tiny_image_success !== 'undefined' && dropZone.tiny_image_success !== null) {
+   		var url = $('#uploadsDiv').children().last().find('img').attr('src');
```

Comment:
Here I just use the last HTML element where I can find the uploaded file's URL. There could be a better way to handle this.

Reply:
looks convoluted indeed

Reply:
I would say it would be an option to send back the url in the response. But that might be a potential data leak. I know you take security very seriously on this project. And I want to make sure of that.